### ISSUE TITLE
Crash on launch of application

### ISSUE THREAD

**Issue by wiverson** [mihaip/infinite-mac#319]
Crash when trying to use Cafe Studio.

Repro steps: trying to use any of the roughly 1995-1997 era configurations. Trying to launch the developer tool Cafe Studio, a simple Java GUI drag and drop tool I wrote back in the day.

Downloaded the Cafe iso, mounting and then copying Cafe Studio to either the main HD or Infinite, neither works. Get the same crash when trying to run other software from the same image as well.

https://www.macintoshrepository.org/12404-symantec-visual-cafe

```
emulator-worker-B9ji8sHX.js:1 RuntimeError: memory access out of bounds
    at SheepShaver.wasm.void powerpc_cpu::execute_loadstore<op_sign_extend_16_32, input_gpr_except<bit_field<11, 15>, 0>, immediate_operand<bit_field<16, 31>, op_sign_extend_16_32>, true, 2, false, false>(unsigned int) (wasm://wasm/SheepShaver.wasm-002c55a2:wasm-function[371]:0x2e117)
    at SheepShaver.wasm.powerpc_cpu::execute(unsigned int) (wasm://wasm/SheepShaver.wasm-002c55a2:wasm-function[278]:0x29b29)
    at SheepShaver.wasm.main (wasm://wasm/SheepShaver.wasm-002c55a2:wasm-function[42]:0x150d1)
    at f._main (https://infinitemac.org/assets/SheepShaver-DddGdz-o.js:3:51001)
    at en (https://infinitemac.org/assets/SheepShaver-DddGdz-o.js:3:51651)
    at e (https://infinitemac.org/assets/SheepShaver-DddGdz-o.js:3:51857)
    at _e (https://infinitemac.org/assets/SheepShaver-DddGdz-o.js:3:51982)
    at r (https://infinitemac.org/assets/SheepShaver-DddGdz-o.js:3:51505)
    at lr (https://infinitemac.org/assets/SheepShaver-DddGdz-o.js:1:3085)
    at e (https://infinitemac.org/assets/SheepShaver-DddGdz-o.js:1:4581)
emulatorDidHaveError @ emulator-worker-B9ji8sHX.js:1
quit @ emulator-worker-B9ji8sHX.js:1
```

[infinitemac.org-1715980795146.log](https://github.com/mihaip/infinite-mac/files/15358281/infinitemac.org-1715980795146.log)

**Comment by dingusdev** [comment]
I was able to get this running on Mac OS 8.6 (Sheepshaver), though I did have to add an Apple Extras folder and move the Apple Software stuff manually.

![image](https://github.com/mihaip/infinite-mac/assets/52434309/52321693-f146-44db-8409-9429dc279073)

On a different note, I decided to run this in DingusPPC (using the 6100 emulation) and it too crashed on a memory access error.

**Comment by wiverson** [comment]
Visual Cafe is the first ISO, the one I'm working on is Cafe (not-visual) and the Cafe Studio.

Here's my steps:

Use Custom, System 8.6.
Power Macintosh 9500 (SheepShaver)
256MB
Screen Size automatic.

Drag and drop the [Cafe.iso from macintoshrepository.org](https://www.macintoshrepository.org/download.php?id=38450) to mount.

Run the installer, select Custom Install, include Symantec Cafe, Cafe Studio, Java Demos to Macintosh HD. Select Restart at the end of the install.

System restarts, everything is in the right place, but launching cafe studio results in first an no printer driver error

<img width="745" alt="image" src="https://github.com/mihaip/infinite-mac/assets/873850/51d3b1f7-b7c8-4600-9aef-502da5603cfa">

And then the emulator memory access out of bounds error.

I was on the team that worked on Visual Cafe, so getting that running is already a big blast to the heart - thank you for that. I wrote Cafe Studio, so that one's a little closer still.

Let me know if you need any more details/any Qs...

**Comment by wiverson** [comment]
Can you clarify the source/target paths for the Apple Extras/Apple Software stuff? Does it matter which drive/folder location for install target?

**Comment by dingusdev** [comment]
For the Apple Extras stuff, I first had to move the Apple Software folder into the Saved HD drive. Then, I created a folder named "Apple Extras" (no quotes) in the System Folder (the location does matter) and followed the instructions left in the folder. I also skipped restarting it the first time.

Also, be warned that any changes to the System folder won't carry over if you decide to close out of the browser window and launch it back up.

**Comment by wiverson** [comment]
Lol I literally was doing the exact same thing and got it working while you were posting this. That Chooser error ftw...

I don't think I can convey what an absolute blast from the past this is... for grins, check out this video of me in 1997... https://www.youtube.com/watch?v=i5mEYPqQVPE ... and here is my current YouTube channel - https://www.youtube.com/@ChangeNode

I'll be editing together footage of these old dev tools, probably including THINK Pascal, THINK C, Metrowerks, Cafe/Cafe Studio, and Visual Cafe. Now everything is working!

Thank you so much for your help!

**Comment by wiverson** [comment]
The easter egg about box lol

<img width="647" alt="image" src="https://github.com/mihaip/infinite-mac/assets/873850/c94a219f-887c-4348-8556-96f24e0da66e">

**Comment by dingusdev** [comment]
I like how the easter egg box is an unencrypted resource too. Definitely couldn't get away with that today. On a final note, here's DingusPPC running Cafe Studio: 
![image](https://github.com/mihaip/infinite-mac/assets/52434309/b44c8c5e-1599-474e-ab1f-dce5e6acad87)

**Comment by wiverson** [comment]
So, here's the story...

There was a huge rush to get the Java tools out. The PC guys had a resource editor that they hacked up over a few weeks to generate Java code. The Mac guys in Boston said it would take months to build something similar. The VP of the group wanted to greenlight industry ads for the product with screenshots, but w/o a Java resource editor there would be no ads for the Mac version.

So, I told my boss that I thought it was ridiculous that it would take months to build the resource editor. He said if I was so smart I should just do it (I had just turned 22). So I said sure, I'll take that bet.

I took a little space ship strategy game I was working on and hacked it up over the weekend and created Cafe Studio. I wrote a little parser, etc. Instead of firing lasers, it would fire events. I brought it in on Monday morning on a floppy and melted the mind of my boss. He called the journalist covering devtools and we demo'd it at MacWeek on Tuesday afternoon. It was on the front cover of MacWeek that Friday, complete with the screenshots I made featuring Nine Inch Nails track titles.

My boss realized that I had built it on the weekend on my own hardware, and so technically he needed me to sign the rights over for a bonus, which I was happy to do and handed over a floppy with the source.

A few weeks later I get a call from this very nice woman in the internationalization team. 

"Are you Will?"
"Yes"
"Did you write Cafe Studio?"
"Yes"
"I think I have the wrong source code... everything in the code is talking about space ships....?"

Those were the days.
